wger/core/api/views.py

```diff
@@ -33,10 +37,78 @@
     DaysOfWeekSerializer,
     LicenseSerializer,
     RepetitionUnitSerializer,
-    WeightUnitSerializer
+    WeightUnitSerializer,
+    UserprofileSerializer,
+    UserCreateSerializer
 )
-from wger.core.api.serializers import UserprofileSerializer
+
 from wger.utils.permissions import UpdateOnlyPermission, WgerPermission
+from wger.config.models import GymConfig
+from wger.gym.models import GymUserConfig
+
+
+class UserCreateViewSet(viewsets.ViewSet):
+    '''
+    API endpoint to create new users
+    '''
+    # protected endpoint
+    is_private = True
+
+    def create(self, request):
+        request_data = JSONParser().parse(request)
+
+        # get user profile
+        user_profile = UserProfile.objects.get(user=request.user)
+        
+        # check if user is allowed to create users via API
+
+        if user_profile.can_create_user:
+            password = request_data.get('password', None)
+            confirm_password = request_data.get('confirm_password', None)
+            email = request_data.get('email', None)
+            username = request_data.get('username', None)
+
+            # check passwords match don't match return 400
+            if password != confirm_password:
+                return Response({'message': 'Passwords provided do not match!'}, status=status.HTTP_400_BAD_REQUEST)
+            
+            # pass request data through serializer
+            user_serializer = UserCreateSerializer(data=request_data)
+            
+            # check if serializer is valid
+            if user_serializer.is_valid():
+                # get creator ID by introspecting token
+                creator = User.objects.get(pk=Token.objects.get(key=request.auth).user_id)
+                data = user_serializer.data
+                username, email, password = data.get('username'), data.get('email', None), data.get('password')
+
+                # create a user object
+                user = User.objects.create_user(username=username, email=email, password=password)
+                user.save()
+
+                # create user profile
+                user.userprofile.creator = creator.username
+                user.userprofile.token = request.auth.key
+                
+                # assign user a gym
+                gym_config = GymConfig.objects.get(pk=1)
+                if gym_config.default_gym:
+                    user.userprofile.gym = gym_config.default_gym
+
+                    # Create gym user configuration object
```

Comment:
Good work with adding comments, It's now easy to follow and understand the code.

Reply:
I like the way u've added comments to the code, It's now easy to follow and understand the code.I also like that users can be added through the API.The problem is that this feature is causing the test coverage to drop by %.Please write more tests for this feature or if you have exhaustively tested it then test other components of the codebase to ensure the coverage does not drop.